x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java

```diff
@@ -922,26 +923,26 @@ public void testServiceAccountDenial() {
         );
         assertThat(
             securityException,
-            throwableWithMessage(
-                containsString("[" + action + "] is unauthorized" + " for " + "service account" + " [" + serviceUser.principal() + "]")
-            )
+            throwableWithMessage(containsString("[" + action + "] is unauthorized for service account [" + serviceUser.principal() + "]"))
         );
-        if (isRunAs) {
-            assertThat(securityException, throwableWithMessage(containsString("run as [" + finalUser.principal() + "] with roles [")));
-        }
-        assertThat(securityException, throwableWithMessage(containsString("this action is granted by the index privileges [read,all]")));
         if (isRunAs && false == canRunAs) {
             verify(auditTrail).runAsDenied(eq(requestId), eq(authentication), eq(action), eq(request), authzInfoRoles(role.names()));
+            assertThat(securityException, throwableWithMessage(containsString("run as [" + finalUser.principal() + "]")));
         } else {
             if (canRunAs) {
                 verify(auditTrail).runAsGranted(eq(requestId), eq(authentication), eq(action), eq(request), authzInfoRoles(role.names()));
+                assertThat(securityException, throwableWithMessage(containsString("run as [" + finalUser.principal() + "] with roles [")));
+                assertThat(
+                    securityException,
+                    throwableWithMessage(containsString("this action is granted by the index privileges [read,all]"))
+                );
```

Comment:
I believe this assertion should be moved outside of this `if (canRunAs)` block and at the same level of the following `verify(...)` call.

Reply:
Currently, the run as denied message does not include context on index privileges since it's somewhat irrelevant context and distracts from the underlying issue (that the run as is not allowed, which is not related to index privs). Would prefer to keep it that way - wdyt?

Reply:
If I read the conditions correctly, outside this if block, the error message should not be about run-as, but just for an unauthorised action.

Reply:
Ahh you're totally right! Thanks